app/code/Magento/Sales/Model/Order.php

```diff
@@ -1199,16 +1200,15 @@ public function getTrackingNumbers()
      * Retrieve shipping method
      *
      * @param bool $asObject return carrier code and shipping method data as object
-     * @return string|\Magento\Framework\DataObject
+     * @return string|ShippingMethod
```

Comment:
This is backward incompatible change. SippingMethod should be subset of DataObject or [new public method should be introduced](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) and this declared as `@deprecated`

Reply:
Extending from DataObject was my first (defensive) approach, but consent seemed to be that it's unnecessary, so I removed it. I'm okay with reverting it though

Reply:
I added back the `DataObject` inheritance, and deprecated the inherited methods as originally intended